test/lib/test_stdext.ml

```diff
@@ -0,0 +1,47 @@
+module Test_string = struct
+  module String = Stdext.String
+
+  let drop_prefix =
+    let make_test string ~prefix ~expected =
+      let test_name =
+        Printf.sprintf "drop_prefix %s ~prefix:%s" string prefix
+      in
+      let test_fun () =
+        let actual = String.drop_prefix string ~prefix in
+        Alcotest.(check (option string) test_name expected actual)
+      in
+      (test_name, `Quick, test_fun)
+    in
+    [
+      make_test "" ~prefix:"" ~expected:(Some "");
+      make_test "foo" ~prefix:"foo" ~expected:(Some "");
+      make_test "foo" ~prefix:"" ~expected:(Some "foo");
+      make_test "" ~prefix:"foo" ~expected:None;
+      make_test "foo" ~prefix:"o" ~expected:None;
+      make_test "foo" ~prefix:"f" ~expected:(Some "oo");
+    ]
+
+  let drop_suffix =
+    let make_test string ~suffix ~expected =
+      let test_name =
+        Printf.sprintf "drop_suffix %s ~suffix:%s" string suffix
+      in
+      let test_fun () =
+        let actual = String.drop_suffix string ~suffix in
+        Alcotest.(check (option string) test_name expected actual)
+      in
+      (test_name, `Quick, test_fun)
+    in
+    [
+      make_test "" ~suffix:"" ~expected:(Some "");
+      make_test "foo" ~suffix:"foo" ~expected:(Some "");
+      make_test "foo" ~suffix:"" ~expected:(Some "foo");
+      make_test "" ~suffix:"foo" ~expected:None;
+      make_test "foo" ~suffix:"o" ~expected:(Some "fo");
+      make_test "foo" ~suffix:"f" ~expected:None;
+    ]
+
+  let tests = List.concat [ drop_prefix; drop_suffix ]
+end
+
+let suite = ("Stdext", Test_string.tests)
```

Comment:
This is nice to detect issues, though maybe we shouldn't go to deep into the alternate stdlib game. It could be better to use `Base.String.val chop_suffix` since we have a direct dependency on `base` anyway. I did a quick check and it looks like it passes the tests exactly.

(I also thought `Astring` which we probably indirectly depend on might have something like that and while it has `Astring.String.is_prefix` exists, I can't find a way to drop the affix)

Reply:
I've switched to using `Base.String.chop_suffix` and `chop_prefix` and removed the now unused `String.drop_suffix`